enterprise/coderd/provisionerkeys.go

```diff
@@ -98,6 +115,50 @@ func (api *API) provisionerKeys(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, convertProvisionerKeys(pks))
 }
 
+// @Summary List provisioner key daemons
+// @ID list-provisioner-key-daemons
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Enterprise
+// @Param organization path string true "Organization ID"
+// @Success 200 {object} []codersdk.ProvisionerKeyDaemons
+// @Router /organizations/{organization}/provisionerkeys/daemons [get]
+func (api *API) provisionerKeyDaemons(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	organization := httpmw.OrganizationParam(r)
+
+	pks, err := api.Database.ListProvisionerKeysByOrganization(ctx, organization.ID)
+	if err != nil {
+		httpapi.InternalServerError(rw, err)
+		return
+	}
+	sdkKeys := convertProvisionerKeys(pks)
+
+	daemons, err := api.Database.GetProvisionerDaemonsByOrganization(ctx, organization.ID)
+	if err != nil {
+		httpapi.InternalServerError(rw, err)
+		return
+	}
+	// provisionerdserver.DefaultHeartbeatInterval*3 matches the healthcheck report staleInterval.
+	recentDaemons := db2sdk.RecentProvisionerDaemons(time.Now(), provisionerdserver.DefaultHeartbeatInterval*3, daemons)
+
+	pkDaemons := make([]codersdk.ProvisionerKeyDaemons, 0, len(sdkKeys))
```

Comment:
if there's more than one provisioner per key on average, this'll reallocate. a safer upper-bound would be `len(recentDaemons)`, but there is the possibility that it could overshoot by a bit. I'm fine with either, just making sure the tradeoffs have been considered!

Reply:
Reallocating is not a big deal.